diff -Naur sheerdns.orig/dir.h sheerdns/dir.h
--- sheerdns.orig/dir.h	2003-04-12 10:04:59.000000000 +0200
+++ sheerdns/dir.h	2010-12-29 20:02:21.118916669 +0100
@@ -12,3 +12,4 @@
 time_t get_mtime (int qtype, unsigned char *s);
 
 #define SHEERDNS_DIR	"/var/sheerdns"
+#define SHEERDNS_CMODE	0770
diff -Naur sheerdns.orig/hash.c sheerdns/hash.c
--- sheerdns.orig/hash.c	2010-12-29 20:01:38.595583335 +0100
+++ sheerdns/hash.c	2010-12-29 20:02:49.248916668 +0100
@@ -28,7 +28,7 @@
 	unsigned char *s;
 	s = (unsigned char *) hex_hash ((unsigned char *) argv[1]);
 	snprintf (buf, sizeof (buf), SHEERDNS_DIR "/%s/%s", s, argv[1]);
-	mkdir (buf, 0700);
+	mkdir (buf, SHEERDNS_CMODE);
 	l = strlen ((char *) s);
 	write (1, s, l);
 	write (1, "\n", 1); }
diff -Naur sheerdns.orig/sheerdns.c sheerdns/sheerdns.c
--- sheerdns.orig/sheerdns.c	2010-12-29 20:01:38.598916669 +0100
+++ sheerdns/sheerdns.c	2010-12-29 20:02:21.118916669 +0100
@@ -583,25 +583,25 @@
     const char *hex = "0123456789ABCDEF";
     char path[256];
     int j, fd, l;
-    mkdir (SHEERDNS_DIR, 0700);
+    mkdir (SHEERDNS_DIR, SHEERDNS_CMODE);
     strcpy (path, SHEERDNS_DIR);
-    mkdir (path, 0700);
+    mkdir (path, SHEERDNS_CMODE);
     l = strlen (path);
     for (j = 0; j < 256; j++) {
 	path[l] = '/';
 	path[l + 1] = hex[j >> 4];
 	path[l + 2] = hex[j & 0xF];
 	path[l + 3] = '\0';
-	mkdir (path, 0700); }
-    mkdir (SHEERDNS_DIR "/C9/localhost", 0700);
-    fd = open (SHEERDNS_DIR "/C9/localhost/A", O_WRONLY | O_CREAT | O_TRUNC, 0700);
+	mkdir (path, SHEERDNS_CMODE); }
+    mkdir (SHEERDNS_DIR "/C9/localhost", SHEERDNS_CMODE);
+    fd = open (SHEERDNS_DIR "/C9/localhost/A", O_WRONLY | O_CREAT | O_TRUNC, SHEERDNS_CMODE);
     write (fd, "127.0.0.1", 9);
     close (fd);
-    mkdir (SHEERDNS_DIR "/7A/localhost.localdomain", 0700);
-    fd = open (SHEERDNS_DIR "/7A/localhost.localdomain/A", O_WRONLY | O_CREAT | O_TRUNC, 0700);
+    mkdir (SHEERDNS_DIR "/7A/localhost.localdomain", SHEERDNS_CMODE);
+    fd = open (SHEERDNS_DIR "/7A/localhost.localdomain/A", O_WRONLY | O_CREAT | O_TRUNC, SHEERDNS_CMODE);
     write (fd, "127.0.0.1", 9);
     close (fd);
-    mkdir (SHEERDNS_DIR "/B5/127.0.0.1", 0700);
-    fd = open (SHEERDNS_DIR "/B5/127.0.0.1/PTR", O_WRONLY | O_CREAT | O_TRUNC, 0700);
+    mkdir (SHEERDNS_DIR "/B5/127.0.0.1", SHEERDNS_CMODE);
+    fd = open (SHEERDNS_DIR "/B5/127.0.0.1/PTR", O_WRONLY | O_CREAT | O_TRUNC, SHEERDNS_CMODE);
     write (fd, "localhost", 9);
     close (fd); }
