diff -Naur sleuthkit-4.4.0.orig/tsk/auto/Makefile.am sleuthkit-4.4.0/tsk/auto/Makefile.am
--- sleuthkit-4.4.0.orig/tsk/auto/Makefile.am	2017-01-18 11:27:05.000000000 +0900
+++ sleuthkit-4.4.0/tsk/auto/Makefile.am	2017-03-18 11:33:27.937020434 +0900
@@ -3,7 +3,8 @@
 
 noinst_LTLIBRARIES = libtskauto.la
 # Note that the .h files are in the top-level Makefile
-libtskauto_la_SOURCES = auto.cpp auto_db.cpp sqlite3.c sqlite3.h db_sqlite.cpp db_postgresql.cpp case_db.cpp guid.cpp tsk_db.cpp tsk_case_db.h tsk_auto.h tsk_auto_i.h tsk_case_db.h tsk_db.h tsk_db_sqlite.h tsk_db_postgresql.h db_connection_info.h guid.h 
+libtskauto_la_SOURCES = auto.cpp auto_db.cpp db_sqlite.cpp db_postgresql.cpp case_db.cpp guid.cpp tsk_db.cpp tsk_case_db.h tsk_auto.h tsk_auto_i.h tsk_case_db.h tsk_db.h tsk_db_sqlite.h tsk_db_postgresql.h db_connection_info.h guid.h 
+libtskauto_la_LIBADD = -lsqlite3
 
 indent:
 	indent *.cpp *.h
diff -Naur sleuthkit-4.4.0.orig/tsk/auto/db_sqlite.cpp sleuthkit-4.4.0/tsk/auto/db_sqlite.cpp
--- sleuthkit-4.4.0.orig/tsk/auto/db_sqlite.cpp	2017-01-18 11:27:05.000000000 +0900
+++ sleuthkit-4.4.0/tsk/auto/db_sqlite.cpp	2017-03-18 11:32:04.077018663 +0900
@@ -14,7 +14,7 @@
 */
 
 #include "tsk_db_sqlite.h"
-#include "sqlite3.h"
+#include <sqlite3.h>
 #include "guid.h"
 #include <string.h>
 #include <sstream>
diff -Naur sleuthkit-4.4.0.orig/tsk/auto/tsk_db_sqlite.h sleuthkit-4.4.0/tsk/auto/tsk_db_sqlite.h
--- sleuthkit-4.4.0.orig/tsk/auto/tsk_db_sqlite.h	2017-01-18 11:27:05.000000000 +0900
+++ sleuthkit-4.4.0/tsk/auto/tsk_db_sqlite.h	2017-03-18 11:34:20.853688218 +0900
@@ -19,7 +19,7 @@
 
 #include <map>
 
-#include "sqlite3.h"
+#include <sqlite3.h>
 #include "tsk_db.h"
 
 using std::map;
